src/lib/operators/jit_operator/operators/jit_read_tuple.cpp

```diff
@@ -0,0 +1,117 @@
+#include "jit_read_tuple.hpp"
+
+#include "constant_mappings.hpp"
+#include "resolve_type.hpp"
+#include "storage/create_iterable_from_column.hpp"
+
+namespace opossum {
+
+std::string JitReadTuple::description() const {
+  std::stringstream desc;
+  desc << "[ReadTuple] ";
+  for (const auto& input_column : _input_columns) {
+    desc << "x" << input_column.tuple_value.tuple_index() << " = Col#" << input_column.column_id << ", ";
+  }
+  for (const auto& input_literal : _input_literals) {
```

Comment:
Do you maintain the orders of the columns somewhere?

Reply:
I am not sure I 100% understand what you mean.
The `column_id` refers to the original column in the input table.
The order of the columns within the tuple used in the operator is arbitrary, but that should not matter, since this order is never exposed.
The order of columns in the output table is determined by the `JitWriteTuple` operator (i.e. how it is configured by the `LQPTranslator`).